subsys/net/l2/canbus/canbus.c

```diff
@@ -347,6 +372,19 @@ static enum net_verdict finish_pkt(struct net_pkt *pkt)
 	net_buf_pull(pkt->buffer, net_pkt_lladdr_dst(pkt)->len +
 				  net_pkt_lladdr_src(pkt)->len);
 
+#if defined(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR)
+	if (is_for_translator(pkt)) {
+		/* Pull room for address extension */
+		net_buf_pull(pkt->buffer, sizeof(struct net_eth_addr) -
+					  net_pkt_lladdr_src(pkt)->len);
+		/* Set the destination address to the inline MAC and pull it */
+		net_pkt_cursor_init(pkt);
+		net_pkt_lladdr_dst(pkt)->addr = net_pkt_cursor_get_pos(pkt);
+		net_pkt_lladdr_dst(pkt)->type = NET_LINK_ETHERNET;
+		net_pkt_lladdr_dst(pkt)->len = sizeof(struct net_eth_addr);
+		net_buf_pull(pkt->buffer, sizeof(struct net_eth_addr));
```

Comment:
Hmm, I think you should init the cursor *after* you have pulled stuff, otherwise the cursor would point to "unexpected" position (unless you want that of course).